custom_components/casambi_bt/__init__.py

```diff
@@ -27,28 +27,22 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Casambi Bluetooth from a config entry."""
-    conf = entry.data
-    casa_api = await async_casmbi_api_setup(
-        hass, conf[CONF_ADDRESS], conf[CONF_PASSWORD]
-    )
-
-    if not casa_api:
-        return False
-
-    casa_api.casa.registerUnitChangedHandler(casa_api._unit_changed_handler)
+    api = CasambiApi(hass, entry.data[CONF_ADDRESS], entry.data[CONF_PASSWORD])
+    await api.connect()
+    api.register_unit_changed_handler()
```

Comment:
This was/is only applied on setup, does it need to be called again on reconnect?

Reply:
No as long as the Casmabi object doesn't change this is fine. The internal callbacks that change with a new connection get abstracted away in the library.

Reply:
Does it require a connection? Otherwise this could go to the init function.

Reply:
No the internal callback required for this callback to fire only gets subscribed to in connect. So register_unit_changed_handler needs to be called before the call to connect (not the order it currently is in)

Reply:
So, should I change it or leave it?